File: activity/templates/app-java/src/main/java/di/components/_Component.java
package <%= appPackage %>.di.components;

import dagger.Component;
import <%= appPackage %>.di.ActivityScope;
import <%= appPackage %>.di.modules.<%= activityName %>Module;
import <%= appPackage %>.ui.<%= activityPackageName %>.<%= activityName %>Activity;

// android-hipster-needle-component-injection-import


@ActivityScope
@Component(dependencies = {ApplicationComponent.class}, modules = {<%= activityName %>Module.class})
public interface <%= activityName %>Component {

    void inject(<%= activityName %>Activity activity);

    // android-hipster-needle-component-injection-method

}

File: activity/templates/app-kotlin/src/main/java/di/components/_Component.kt
package <%= appPackage %>.di.components

import dagger.Component;
import <%= appPackage %>.di.ActivityScope;
import <%= appPackage %>.di.modules.<%= activityName %>Module;
import <%= appPackage %>.ui.<%= activityPackageName %>.<%= activityName %>Activity;

// android-hipster-needle-component-injection-import

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(<%= activityName %>Module::class))
interface <%= activityName %>Component {

    fun inject(activity: <%= activityName %>Activity)

    // android-hipster-needle-component-injection-method

}
